python/ray/tests/test_failure.py

```diff
@@ -735,6 +735,36 @@ def func():
     caplog.clear()
 
 
+def test_transient_error_retry(monkeypatch, ray_start_cluster):
```

Comment:
This test covers the case mentioned in the PR description, and it’s much more aggressive.